syntax = "proto2";

enum DBCommandType {
    GET = 1;
    SET = 2;
    DROP = 3;
}

message DBCommand {
    required DBCommandType type = 1;
    required string key = 2;
    optional string value = 3;
}

enum ResultType {
    OK = 1;
    KEY_MISSING = 2;
    ERROR = 3;
}

message DBCommandResult {
    required ResultType type = 1;
    optional string value = 2;
    optional string error_msg = 3;
}
